le-taxi-api-node.js/src/features/inquiry/inquiry.controller.ts

```diff
@@ -38,54 +49,78 @@ class InquiryController {
       }))
     );
 
-    sendResponse(response, now, () =>
-      closestTaxis.map((closestTaxi, i) => {
-        const departureTime = addSec(
-          now,
-          routesFromTaxiPositionToCustomer[0][i] +
-            configs.taxiRegistryOsrmApi.estimation.biasInSec +
-            configs.taxiRegistryOsrmApi.estimation.requestAndDispatchInSec
-        );
-        const arrivalTime = addSec(
-          departureTime,
-          routeFromSourceToDestination[0].legs[0].duration + configs.taxiRegistryOsrmApi.estimation.biasInSec
-        );
-
-        return {
-          mainAssetType: {
-            id: toMainAssetType((closestTaxi.taxi as any).assetType, closestTaxi.taxi.operatorPublicId)
-          },
-          departureTime,
-          arrivalTime,
-          from,
-          to,
-          pricing: {
-            estimated: true,
-            parts: [
-              {
-                optimisticAmount: 0,
-                amount: 0,
-                pessimisticAmount: 0,
-                currencyCode: 'CAD'
-              }
-            ]
-          }
-        };
-      })
-    );
+    const options = await Promise.all(closestTaxis.map((closestTaxi, i) => toInquiryResponseOptionsDTO({
+      now,
+      taxi: closestTaxi.taxi,
+      from,
+      to,
+      waitDuration: routesFromTaxiPositionToCustomer[0][i],
+      tripDuration: routeFromSourceToDestination[0].legs[0].duration,
+    })));
+
+    sendResponse(response, now, options);
   }
 }
 
-function sendResponse(response: Response, now: string, callback?: () => InquiryResponseOptionsDTO[]) {
+function sendResponse(response: Response, now: string, options?: InquiryResponseOptionsDTO[]) {
   const inquiryResponse = {
     validUntil: addMinutes(now, 5),
-    options: callback ? callback() : []
+    options: options ?? []
   };
 
   response.status(StatusCodes.OK);
   response.json(inquiryResponse);
 }
 
+async function toInquiryResponseOptionsDTO(params: IInquiryResponseParams): Promise<InquiryResponseOptionsDTO> {
+  const assetType = (params.taxi as any).assetType;
```

Comment:
Unfortunately we can't. The assetType field is injected by the mongo query to simplify the results => https://github.com/VilledeMontreal/montreal-taxi-registry/blob/feature/TM-1790-gofs-preparation/le-taxi-api-node.js/src/features/latestTaxiPositions/latestTaxiPosition.repository.ts#L42. This allows us to use the field directly without having to deal with a dictionary and extract the keys since we are using a facet https://www.mongodb.com/docs/manual/reference/operator/aggregation/facet/